runelite-client/src/main/java/net/runelite/client/plugins/party/PartyStatusOverlay.java

```diff
@@ -0,0 +1,173 @@
+import net.runelite.client.party.PartyMember;
+import net.runelite.client.party.PartyService;
+import net.runelite.client.plugins.party.data.PartyData;
+import net.runelite.client.ui.FontManager;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+import net.runelite.client.ui.overlay.OverlayUtil;
+import net.runelite.client.util.ColorUtil;
+
+@Singleton
+public class PartyStatusOverlay extends Overlay
+{
+	
+	private static final Color COLOR_HEALTH_MAX = Color.green;
+	private static final Color COLOR_HEALTH_MIN = Color.red;
+	private static final Color COLOR_PRAYER = new Color(50, 200, 200);
+	private static final Color COLOR_STAMINA = new Color(160, 124, 72);
+	private static final Color COLOR_SPEC = new Color(3, 153, 0);
+	private static final Font OVERLAY_FONT = FontManager.getRunescapeBoldFont().deriveFont(20f);
+
+	private final Client client;
+	private final PartyConfig config;
+	private final PartyService partyService;
+	private final PartyPluginService partyPluginService;
+
+	private BufferedImage vengIcon = null;
+
+	private boolean renderHealth = false;
+	private boolean renderPrayer = false;
+	private boolean renderStamina = false;
+	private boolean renderSpec = false;
+	private boolean renderVeng = false;
+	private boolean renderSelf = false;
+	
+	@Inject
+	public PartyStatusOverlay(
+		Client client, SpriteManager spriteManager,
+		PartyConfig config, PartyService partyService, PartyPluginService partyPluginService
+	)
+	{
+		this.client = client;
+		this.partyService = partyService;
+		this.partyPluginService = partyPluginService;
+		this.config = config;
+
+		spriteManager.getSpriteAsync(SpriteID.SPELL_VENGEANCE_OTHER, 0, bi -> this.vengIcon = bi);
+		updateConfig();
+
+		setLayer(OverlayLayer.ABOVE_SCENE);
+		setPosition(OverlayPosition.DYNAMIC);
+		setPriority(OverlayPriority.HIGH);
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (!partyService.isInParty())
+		{
+			return null;
+		}
+
+		List<PartyMember> partyMembers = partyService.getMembers();
+		for (Player player : client.getPlayers())
+		{
+			if (!renderSelf && player == client.getLocalPlayer())
+			{
+				continue;
+			}
+			
+			PartyMember partyMember = findPartyMember(partyMembers, player);
+			if (partyMember == null)
+			{
+				continue;
+			}
+
+			PartyData partyData = partyPluginService.getPartyData(partyMember.getMemberId());
+			if (partyData == null)
+			{
+				continue;
+			}
+
+
+			graphics.setFont(OVERLAY_FONT);
+			if (renderHealth)
+			{
+				double healthRatio = Math.min(1.0, (double) partyData.getHitpoints() / partyData.getMaxHitpoints());
+				Color healthColor = ColorUtil.colorLerp(COLOR_HEALTH_MIN, COLOR_HEALTH_MAX, healthRatio);
+				renderPlayerOverlay(graphics, player, String.valueOf(partyData.getHitpoints()), healthColor, -20, -20);
```

Comment:
It would be neat to draw these with offsets similar to hitsplats where the offsets depend on number of hitsplats